src/services/nft-chain-service.ts

```diff
@@ -0,0 +1,83 @@
+import {BigNumberish} from '@ethersproject/bignumber'
+import {AddressZero} from '@ethersproject/constants'
+import {Contract} from '@ethersproject/contracts'
+import {JsonRpcProvider, StaticJsonRpcProvider, Web3Provider} from '@ethersproject/providers'
+import {BytesLike, ethers} from 'ethers'
+
+import {GasFee} from '../types/interfaces'
+import {waitForTransactionComplete} from '../utils/contracts'
+import {sleep} from '../utils/utils'
+import CoreChainService from './core-chain-service'
+// import {NetworkMonitor} from '../utils/network-monitor'
+
+class NFTChainService extends CoreChainService {
+  cxipNFT: Contract
+
+  constructor(
+    provider: JsonRpcProvider | StaticJsonRpcProvider | Web3Provider,
+    wallet: ethers.Wallet,
+    chainId: number,
+    address: string,
+    contract?: Contract,
+  ) {
+    super(provider, wallet, chainId)
+    this.cxipNFT = contract ? contract : this.getCxipNFT(address)
+  }
+
+  isExists = async (tokenId: number | string): Promise<boolean> => {
+    return this.cxipNFT.exists(tokenId)
+  }
+
+  getTokenId = async (account: string, blockNumber: number): Promise<BigNumberish> => {
+    // eslint-disable-next-line new-cap
+    const filter = this.cxipNFT.filters.Transfer(AddressZero, account, null)
+    const logs = await this.cxipNFT.queryFilter(filter, blockNumber, 'latest')
+    if (logs.length === 0) {
+      await sleep(1000)
+      return this.getTokenId(account, blockNumber)
+    }
+
+    return (logs[logs.length - 1] as any).args[2]
+  }
+
+  estimateGasForMintingNFT = async (tokenURI?: string): Promise<GasFee> => {
+    const gasPrice = await this.getChainGasPrice()
+    const gasLimit = await this.cxipNFT.estimateGas.cxipMint(0, 1, tokenURI || '')
```

Comment:
The second parameter is `TokenUriType` in the smart contracts is an ENUM. Can you define it so its a bit clearer. Given that you set it to `1`, then we are assuming an IPFS url always. Should this not be part of the function call? We can default or maybe detect it if the string gets passed in.